build-tools/packages/build-cli/src/codeCoverage/getCoverageMetrics.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import JSZip from "jszip";
+import { Parser } from "xml2js";
+import type { CommandLogger } from "../logging.js";
+
+/**
+ * The type for the coverage report, containing the line coverage and branch coverage(in percentage)
+ */
+export interface CoverageMetric {
+	lineCoverage: number;
+	branchCoverage: number;
+}
+
+interface TXmlCoverageReportSchema {
+	coverage: {
+		packages: TXmlCoverageReportSchemaForPackage[];
+	};
+}
+
+interface TXmlCoverageReportSchemaForPackage {
+	package: [
+		{
+			"$": {
+				name: string;
+				"line-rate": string;
+				"branch-rate": string;
+			};
+		},
+	];
+}
+
+const extractCoverageMetrics = (
+	xmlForCoverageReportFromArtifact: TXmlCoverageReportSchema,
+): Map<string, CoverageMetric> => {
+	const report: Map<string, CoverageMetric> = new Map();
+	const coverageForPackagesResult =
+		xmlForCoverageReportFromArtifact.coverage.packages[0]?.package;
+
+	for (const coverageForPackage of coverageForPackagesResult) {
+		const packagePath = coverageForPackage.$.name;
```

Comment:
It may be too much to paste in here, but having some example XML alongside this code might be really helpful, especially since things like `$` in the property names are a bit odd (guessing this comes from the XML in some way).

Reply:
Yes, in their structure, "$" is a property. I think XmlCoverageReportSchema and XmlCoverageReportSchemaForPackage interface should clearly explains the schema in which coverage metric is present.